dotCMS/src/main/java/com/dotmarketing/velocity/directive/DotParse.java

```diff
@@ -108,17 +111,27 @@ String resolveTemplatePath(final Context context, final Writer writer, final Ren
 
 
       return asset.getFileAsset().getAbsolutePath();
-    } catch (Exception e) {
-      Logger.warn(this.getClass(), " - unable to resolve " + templatePath + " getting this: "+ e.getMessage() );
-      if(e.getStackTrace().length>0)
-        Logger.warn(this.getClass(), " - at " + e.getStackTrace()[0]);
-      
-      //If we didn't find the resource don't change the exception type
-      if( e instanceof ResourceNotFoundException ) {
-        throw (ResourceNotFoundException) e;
-      }
-      
-      throw new DotStateException(e);
+    } 
+    catch (ResourceNotFoundException e) {
+        Logger.warn(this.getClass(), " - unable to resolve " + templatePath + " getting this: "+ e.getMessage() );
+        if(e.getStackTrace().length>0){
+          Logger.warn(this.getClass(), " - at " + e.getStackTrace()[0]);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal " - at " 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)